### PR TITLE
add samesite attribute to tomcat cookie

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]

##### Description of change

https://github.com/openequella/openEQUELLA/issues/1458

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/src/com/tle/tomcat/service/impl/TomcatServiceImpl.java [hunk 1]
```diff
@@ -179,6 +180,11 @@ public void startup() {
       context.addErrorPage(accessDenied);
       context.addErrorPage(throwableError);
 
+      Rfc6265CookieProcessor cookieProcessor = new Rfc6265CookieProcessor();
+      cookieProcessor.setSameSiteCookies("none");
+      context.setCookieProcessor(cookieProcessor);
+
+
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/src/com/tle/tomcat/service/impl/TomcatServiceImpl.java]:
```suggestion
```

**Comment by PenghaiZhang** [comment]
Since the changes are simple, I just copied these three line and ran in my local environment. 
What happened was I couldn't login. When I clicked the login button, the page did not jump to home page. Instead, it still opened the login page.
I found the login request, which should be a post request, was somehow changed to a get post.

**Comment by edalex-ian** [comment]
@mrblippy seeing 80b322e is just at the connector configuration stage, I don't believe this will work for setups with oEQ just using HTTP behind a load balancer or proxy (which are then doing the HTTPS). Thoughts?

File: Source/Plugins/Core/com.equella.core/src/com/tle/tomcat/service/impl/TomcatServiceImpl.java [hunk 2]
```diff
@@ -199,6 +200,10 @@ public void startup() {
         connector.setPort(httpsPort);
         connector.setSecure(true);
         setConnector(connector);
+
+        Rfc6265CookieProcessor cookieProcessor = new Rfc6265CookieProcessor();
+        cookieProcessor.setSameSiteCookies("none");
```
[the remark this answers is not in the document]
Comment (abidingotter) [reply Source/Plugins/Core/com.equella.core/src/com/tle/tomcat/service/impl/TomcatServiceImpl.java]: Never mind, I just read the Tomcat source and they do a case insensitive match

**Comment by edalex-ian** [comment]
> > @mrblippy seeing [80b322e](https://github.com/openequella/openEQUELLA/commit/80b322e2faab4c3c7389fec962ffe46b0c9ae71d) is just at the connector configuration stage, I don't believe this will work for setups with oEQ just using HTTP behind a load balancer or proxy (which are then doing the HTTPS). Thoughts?
> 
> @edalex-ian I'm fairly certain proxies / balancers are meant to be using the HTTPS port on Tomcat (even though they handle the actual SSL decoding etc). That way Tomcat believes it's a secure connection even though it isn't. (it just has setSecure(true) on it)

From digging in the code, this has been solved by adding a valve. Indeed in our cluster and elsewhere we only run the HTTP connector, but then we configure the X-Forwarded-For headers in the load balancer and set the config option in oEQ. This changed with #391 I believe.

Also, @PenghaiZhang is now taking a different approach (which we've seen done in places like Shibboleth) of adding a Filter - https://github.com/PenghaiZhang/openEQUELLA/commit/d3966cf06488667a97ef9b8db68fcf86f284d819

**Comment by abidingotter** [comment]
> From digging in the code, this has been solved by adding a valve. Indeed in our cluster and elsewhere we only run the HTTP connector, but then we configure the X-Forwarded-For headers in the load balancer and set the config option in oEQ. This changed with #391 I believe.
> Also, @PenghaiZhang is now taking a different approach (which we've seen done in places like Shibboleth) of adding a Filter - PenghaiZhang@d3966cf

Load balancers should be forwarding all HTTPS requests to the HTTPS connector.  The X-Forwarded-For won't make Tomcat believe it's secure.

**Comment by abidingotter** [comment]
Ok, judging from Jolse's comment it seems that does know it's secure?  I'll do some reading :)

**Comment by edalex-ian** [comment]
After lots of discussions we're now addressing this via #1585 . That's for getting use started @mrblippy with this one. :+1:

**Pull request closed without merging**